sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -437,61 +437,72 @@ object DataSourceStrategy {
     }
   }
 
+  /**
+   * Find the column name of an expression that can be pushed down.
+   */
+  private[sql] def pushDownColName(e: Expression): Option[String] = {
+    def helper(e: Expression): Option[Seq[String]] = e match {
+      case a: Attribute => Some(Seq(a.name))
+      case _ => None
+    }
+    helper(e).flatMap(_.headOption)
```

Comment:
Although I know the background, shall we write like the following simpler way in this PR?
```scala
def helper(e: Expression): Option[String] = e match {
  case a: Attribute => Some(a.name)
  case _ => None
}
helper(e)
```